replica/server.go

```diff
@@ -159,25 +160,45 @@ func (s *Server) isExtentSupported() error {
 	return file.Close()
 }
 
+func (s *Server) initUUID() error {
+	var err error
+	r := &Replica{
+		dir:              s.Dir,
+		ReplicaStartTime: StartTime,
+		mode:             types.INIT,
+	}
+	if r.info, err = ReadInfo(s.Dir); err != nil {
+		return err
+	}
+	if r.info.UUID == "" {
+		UUID := guuid.New().String() + s.ReplicaAddress
+		hash := sha1.New()
+		hash.Write([]byte(UUID))
+		r.info.UUID = fmt.Sprintf("%x", hash.Sum(nil))
+		r.writeVolumeMetaData(r.info.Dirty, r.info.Rebuilding)
```

Comment:
While upgrade/rebuilding, will the information in the master replica be set to Dirty and Rebuilding?

Reply:
This function will be fetching whatever is already written on disk, updating UUID and rewriting the same.
Controller and replicas won't know that they are being upgraded so the states do not change during upgrade.
During rebuilding, master replica state is dirty and state of replica being rebuilt is set to rebuilding.

Reply:
cool! #TIL